driver/pace/driver/state.py

```diff
@@ -189,24 +297,29 @@ def _restart_driver_state(
     driver_grid_data = pace.util.grid.DriverGridData.new_from_metric_terms(metric_terms)
     dycore_state = fv3core.DycoreState.init_zeros(quantity_factory=quantity_factory)
     backend_uses_gpu = is_gpu_backend(dycore_state.u.metadata.gt4py_backend)
-    dycore_state = _overwrite_state_from_restart(
-        path,
-        rank,
-        dycore_state,
-        "restart_dycore_state",
-        backend_uses_gpu,
-    )
+
+    if fortran_data is True:
+        dycore_state = _overwrite_state_from_fortran_restart(
+            path,
+            communicator,
+            dycore_state,
+            "restart_dycore_state",
+            backend_uses_gpu,
+        )
+    else:
+        dycore_state = _overwrite_state_from_restart(
+            path,
+            rank,
+            dycore_state,
+            "restart_dycore_state",
+            backend_uses_gpu,
+        )
+
     active_packages = ["microphysics"]
     physics_state = pace.physics.PhysicsState.init_zeros(
         quantity_factory=quantity_factory, active_packages=active_packages
     )
-    physics_state = _overwrite_state_from_restart(
-        path,
-        rank,
-        physics_state,
-        "restart_physics_state",
-        backend_uses_gpu,
-    )
```

Comment:
Why is this removed?

Reply:
We fill in dycore state from restart data. Elynn said that the physics state gets filled in appropriately from dycore, so no need to write into it

Reply:
Yes, I realized that `copy_dycore_to_physics` means physics state always gets reset from dycore state. So we do not even need to write out physics restart in the pace restart format.